src/components/Overview/PlotlyCharts/ForceDirectedGraph.tsx

```diff
@@ -1,18 +1,181 @@
+    'Company that shared information with facebook about your activity on their website/application',
+    'Apps that have been installed on your mobile device',
+    'Account linked to your Facebook profile',
+    'Game you played on Facebook',
+  ];
+
+  // Filter data object based on the list defined above
+  let filteredData = [];
+  for (let i = 0; i < avastarParsedData.length; i++) {
+    if (actions.includes(avastarParsedData[i].action_type)) {
+      filteredData.push(avastarParsedData[i]);
+    }
+  }
+
+  // Pick randomly 150 data points coming from the filtered data object
+  // @ts-ignore
+  var data = filteredData.sort(() => 0.5 - Math.random()).slice(0, 150);
+
+  // Transform filtered data object into the right shape for the force graph (nodes and links)
+  const setForceGraphData = () => {
+    // Set empty arrays
+    var forceGraphData = [] as any;
+    var nodes = [] as any;
+    var links = [] as any;
+    // Set the initial nodes (You, Facebook, Google)
+    nodes.push({
+      id: 'You',
+      group: 1,
+      color: '#ff6961',
+    });
+    nodes.push({
+      id: 'facebook',
+      group: 2,
+      color: '#59adf6',
+    });
+    nodes.push({
+      id: 'google',
+      group: 3,
+      color: '#42d6a4',
+    });
+    // Set the initial links (You - Facebook, You - Google)
+    links.push({
+      source: 'You',
+      target: 'facebook',
+      value: 10, // intensity of the moving particles defined relatively by the volume of data transferred
+    });
+    links.push({
+      source: 'You',
+      target: 'google',
+      value: 10,
+    });
+
+    // Define function to transform every object that have been filtered
+    const transformData = (
+      initialObject: AvastarParsedDataPoint[],
+      nodesObject: object[],
+      linksObject: object[],
+      forceGraphInput: object[],
+      actionType: string,
+      groupId: number,
+      colorCode: string
+    ) => {
+      for (let i = 0; i < initialObject.length; i++) {
+        if (initialObject[i].action_type === actionType) {
+          nodesObject.push({
+            id: data[i].details,
+            group: groupId,
+            color: colorCode,
+          });
+          linksObject.push({
+            source: 'You',
+            target: data[i].details,
+            value: 3,
+          });
+          linksObject.push({
+            source: data[i].details,
+            target: data[i].platform,
+            value: 3,
+          });
+        }
+      }
+      forceGraphInput.push({
+        nodes: nodesObject,
+        links: linksObject,
+      });
+      return forceGraphInput;
+    };
+
+    // Define group numbers to identify unique categories of companies
+    // 4 = Companies who paid to display you advertising content
+    // 5 = Companies that collected data about you outside the apps you installed
+    // 6 = Apps and games you installed/played
+    // 7 = Companies connected to Facebook and Google
+    const groups = [4, 4, 5, 6, 7, 6];
```

Comment:
Pourquoi tes groupes ne commencent pas par 1 ?

Reply:
Car je définis des nodes "initiaux" qui seront toujours là même si le user n'upload pas de data : 
Node "You" = 1 ; 
Node "Facebook" = 2 ; 
Node "Google" = 3 .

Reply:
Ok d'accord!